File: app/src/main/java/net/osdn/gokigen/a01d/camera/sony/wrapper/eventlistener/ICameraEventObserver.java
package net.osdn.gokigen.a01d.camera.sony.wrapper.eventlistener;

import androidx.annotation.NonNull;

public interface ICameraEventObserver
{
    void activate();
    boolean start();
    void stop();
    void release();

    void setEventListener(@NonNull ICameraChangeListener listener);
    void clearEventListener();

    ICameraStatusHolder getCameraStatusHolder();
}

File: app/src/main/java/net/osdn/gokigen/a01d/camera/panasonic/wrapper/eventlistener/ICameraEventObserver.kt
package net.osdn.gokigen.a01d.camera.panasonic.wrapper.eventlistener

import net.osdn.gokigen.a01d.camera.ICameraChangeListener

interface ICameraEventObserver
{
    fun activate()
    fun start(): Boolean
    fun stop()
    fun release()

    fun setEventListener(listener: ICameraChangeListener)
    fun clearEventListener()

    fun getCameraStatusHolder(): ICameraStatusHolder
}
